server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -405,8 +368,32 @@ void executeRequest(
                     searchPhaseProvider.apply(delegate)
                 );
             } else {
+                if ((listener instanceof TelemetryListener tl) && CCS_TELEMETRY_FEATURE_FLAG.isEnabled()) {
+                    tl.setRemotes(resolvedIndices.getRemoteClusterIndices().size());
+                    if (isAsyncSearchTask(task)) {
+                        tl.setFeature(CCSUsageTelemetry.ASYNC_FEATURE);
+                    }
+                    String client = task.getHeader(Task.X_ELASTIC_PRODUCT_ORIGIN_HTTP_HEADER);
```

Comment:
Asking to learn - did we confirm with Kibana or the devtools team that this is the place they will set their client info?

Reply:
I checked the source code and the flag is set there. I can also verbally confirm with them too.